File: stock-serveur-v2/src/main/java/cm/belrose/stockserveur/model/Fournisseur.java
package cm.belrose.stockserveur.model;

import cm.belrose.stockserveur.config.audit.Auditable;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import org.hibernate.envers.NotAudited;

import javax.persistence.*;
import java.io.Serializable;
import java.time.Instant;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "fournisseurs")
public class Fournisseur extends Auditable<String> implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nom;
    private String prenom;
    private String genre;
    private Instant dateNaissance;
    private String lieuNaissance;
    private String email;
    private String numeroTelephone;
    private String photo;
    @Embedded
    private Adresse adresse;
    //Ce champs est juste mis pour simplifier les choses
    private Long entrepriseId;
    @NotAudited
    @OneToMany(mappedBy = "fournisseur")
    private List<CommandeFournisseur> commandeFournisseurs;
}

File: stock-client-v2/src/gs-api/src/models/fournisseur.ts
/* tslint:disable */
import { Adresse } from './adresse';
import { CommandeFournisseur } from './commande-fournisseur';
export interface Fournisseur {
  createdBy?: string;
  updatedBy?: string;
  createdDate?: string;
  updateDate?: string;
  id?: number;
  nom?: string;
  prenom?: string;
  genre?: string;
  dateNaissance?: number;
  lieuNaissance?: string;
  email?: string;
  numeroTelephone?: string;
  photo?: string;
  adresse?: Adresse;
  entrepriseId?: number;
  commandeFournisseurs?: Array<CommandeFournisseur>;
}
